### PR TITLE
pubsub: change Subscriber defaults

### PR DESCRIPTION
**Pull request by pongad**
@mdietz94 @jadekler

### CODE REVIEW

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java [hunk 1]
```diff
@@ -500,7 +490,31 @@ Builder setInternalHeaderProvider(HeaderProvider internalHeaderProvider) {
...
+     * <ul>
+     *   <li>there are at most ten thousand outstanding messages, and
+     *   <li>the combined size of outstanding messages does not exceed 1GB.
+     * </ul>
+     *
+     * "Outstanding messages" here means the messages that have already been given to {@link
+     * MessageReceiver} but not yet {@code acked()} or {@code nacked()}.
+     *
+     * <pre>{@code
+     *    FlowControlSettings flowControlSettings =
```
Comment (jadekler) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: nit: the spacing is a bit awkward in this pre block - could it be made consistent?

File: CHANGES.md [hunk 2]
```diff
@@ -34,3 +35,19 @@
   - `subscription.getTopicAsTopicNameOneof()`: use `TopicNames.parse(subscription.getTopic())`
   - `subscription.getNameAsSubscriptionName()`: use `ProjectSubscriptionName.parse(subscription.getName())`
   - `snapshot.getNameAsSnapshotName()`: use `ProjectSnapshotName.parse(snapshot.getName())`
+
+# v0.44.0
```
Comment (jadekler) [inline CHANGES.md]: What happened to v0.37.0 and so on? o.O

**Comment by jadekler** [comment]
LGTM!

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java (hunk 1 above)
Comment (pongad) [reply google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: Ah thank you for catching!

File: CHANGES.md (hunk 2 above)
Comment (pongad) [reply CHANGES.md]: The smaller notes are written [here](https://github.com/GoogleCloudPlatform/google-cloud-java/releases). This file is usually for notes with significant explanations, etc.

**Pull request merged**